doc/design/csp.md

```diff
@@ -89,6 +89,55 @@ The point here is that we need a consistent way to compose types, like in C++ we
 
 ### Select
 
+In Go, the `select` statement lets a goroutine wait on multiple communication operations. A `select` blocks untill one of its cases can run, then it executes that case. It chooses one at random if multiple are ready.
+
+```go
+
+ch1  := make(chan int)       
+ch2  := make(chan int, 100)
+
+x := 0
+
+for {
+    select {
+    case ch1 <- x:
+      x := x + 1
+    case y <- ch2:
+      fmt.Println("Received on channel")
+    default:
+      fmt.Println("Default")
+    }
+  }
+
+```
+
+In Fluid, we should be able to do the same:
+
+```python
+ch1  = fluid.make_chan(dtype=INT)
+ch2 = fluid.make_chan(dtype=INT, 100)
+
+select = fluid.select()
+
+with sel.case(ch1, 'w', X):
```

Comment:
I think we need a way to differentiate the following two cases:

```Go
  select {
    case ch1 <- x:
      x := x + 1
    case y <- ch2:
      fmt.Println("Received on channel")
  }
```

```Go
  select {
    case ch1 <- x:
      x := x + 1
  }

  select {
    case y <- ch2:
      fmt.Println("Received on channel")
   }
```

The first one means wait for either of the case being true, and continue. The second one means wait for the first to be true, and then wait for the second to be true. But I think the proposed Python syntax can't differentiate between the two case.

Reply:
I agree with this as well, I think the default case in `select` should be the first example i.e. wait for either of the cases to unblock and if multiple cases unblock at the same time, we decide which block to execute pseudo-randomly. Otherwise execute the case that unblocked first.

Reply:
Sorry my mistake. There is a typo in the code. The code should be `sel =fluid.select()`. This way, the code can differentiate between the two cases. For the first case the code will be
```
sel1 = fluid.select()
with sel1.case(ch1, 'w', x):
    ...
    ...

with sel1.case(ch2, 'r', y):
    ...
    ...
```

For the second case the code will be like this 

```
sel1 = fluid.select()
with sel1.case(ch1, 'w', x):
    ...
    ...

sel2 = fluid.select()
with sel2.case(ch2, 'r', y):
    ...
    ...
```